internal/cmd/beta/security-group/delete/delete.go

```diff
@@ -0,0 +1,107 @@
+package delete
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/spf13/cobra"
+	"github.com/stackitcloud/stackit-cli/internal/pkg/args"
+	"github.com/stackitcloud/stackit-cli/internal/pkg/errors"
+	"github.com/stackitcloud/stackit-cli/internal/pkg/examples"
+	"github.com/stackitcloud/stackit-cli/internal/pkg/globalflags"
+	"github.com/stackitcloud/stackit-cli/internal/pkg/print"
+	"github.com/stackitcloud/stackit-cli/internal/pkg/projectname"
+	"github.com/stackitcloud/stackit-cli/internal/pkg/services/iaas/client"
+	"github.com/stackitcloud/stackit-cli/internal/pkg/utils"
+	"github.com/stackitcloud/stackit-sdk-go/services/iaas"
+)
+
+type inputModel struct {
+	*globalflags.GlobalFlagModel
+	SecurityGroupId string
+}
+
+const groupIdArg = "GROUP_ID"
+
+func NewCmd(p *print.Printer) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "delete",
+		Short: "Deletes a security group",
+		Long:  "Deletes a security group by its internal ID.",
+		Args:  args.SingleArg(groupIdArg, utils.ValidateUUID),
+		Example: examples.Build(
+			examples.NewExample(`Delete a named group with ID "xxx"`, `$ stackit beta security-group delete xxx`),
+		),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			ctx := context.Background()
+			model, err := parseInput(p, cmd, args)
+			if err != nil {
+				return err
+			}
+
+			// Configure API client
+			apiClient, err := client.ConfigureClient(p)
+			if err != nil {
+				return err
+			}
+
+			projectLabel, err := projectname.GetProjectName(ctx, p, cmd)
+			if err != nil {
+				return fmt.Errorf("get project name: %w", err)
+			}
+
+			securityGroupResp, err := apiClient.GetSecurityGroup(ctx, model.ProjectId, model.SecurityGroupId).Execute()
```

Comment:
instead of doing this here, as you will also need this functionality in the `update` command, please create a utils method in `internal/pkg/services/iaas/utils.go`. You'll find very similar utils there for the other resources

And then you should also apply the same logic as suggested above, instead of failing the command in case fetching the security group name fails, we show the ID

Reply:
s. above: introduced a suitable util function